modules/acquia_telemetry/src/Event.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Drupal\acquia_telemetry;
+
+use JsonSerializable;
+
+/**
+ * An event object as defined by Amplitude API specifications.
+ *
+ * @see https://developers.amplitude.com/#http-api
+ */
+class Event implements JsonSerializable {
+
+  /**
+   * The event type.
+   *
+   * @var string
+   */
+  protected $type;
+
+  /**
+   * An array of event data.
+   *
+   * This key value pair will be sent to Amplitude as event_properties.
+   *
+   * @var array
+   */
+  protected $data;
+
+  /**
+   * The Amplitude user ID for the event.
+   *
+   * We do not use actual user ID. Instead, the UUID of the application is
+   * used to uniquely identify the event while preserving anonymity.
+   *
+   * @var string
+   */
+  protected $user_id;
+
+  /**
+   * Event constructor.
+   *
+   * @param string $type
+   *   The event type.
+   * @param array $data
+   *   The event properties.
+   */
+  public function __construct($type, $user_id, $data = []) {
```

Comment:
The doc comment does not mention $user_id, and $data should be type hinted as an array.